os/mm/mm_heap/mm_extend.c

```diff
@@ -106,6 +107,7 @@ void mm_extend(FAR struct mm_heap_s *heap, FAR void *mem, size_t size, int regio
 	/* Take the memory manager semaphore */
 
 	mm_takesemaphore(heap);
+	sched_lock();
```

Comment:
I think sched_lock should be before taking semaphore.
Otherwise, there might be a context switching between taking semaphore and sched_lock.

Reply:
In addition, when sched_lock() is used, is a semaphore still required?